packages/block-library/src/gallery/gallery-image.native.js

```diff
@@ -128,15 +128,15 @@ class GalleryImage extends Component {
 	}
 
 	finishMediaUploadWithSuccess( payload ) {
-		this.props.setAttributes( {
-			id: payload.mediaServerId,
-			url: payload.mediaUrl,
-		} );
-
 		this.setState( {
 			isUploadInProgress: false,
 			didUploadFail: false,
 		} );
+
+		this.props.setAttributes( {
+			id: payload.mediaServerId,
+			url: payload.mediaUrl,
+		} );
```

Comment:
I read about this in your post, in cases when we want to do something after a `setState` happens you could use the second parameter which is a callback that will trigger after. like:

```
	this.setState( {
		isUploadInProgress: false,
		didUploadFail: false,
	}, () => {
		this.props.setAttributes( {
			id: payload.mediaServerId,
			url: payload.mediaUrl,
		} );
	} );
```

Reply:
Hi @geriux :wave:  Thanks for testing and reviewing!

> you could use the second parameter which is a callback

Thanks for noting this. Using the callback is useful to guarantee that `setAttributes` is called after the state update has been applied. In this particular case, we don't need to read `this.state` values for use in `setAttributes`, so we don't actually need that guarantee. Instead, the problem was that we couldn't call `setState` _after_ `setAttributes`, since invoking `setAttributes` results in the component becoming unmounted. Since we are also using `componentDidUpdate` with this component, I don't think the callback is necessary.

Reply:
Nice! Sounds good!

Reply:
Thanks!